cfme/automate/explorer/namespace.py

```diff
@@ -90,8 +90,13 @@ def create(self, name=None, description=None, cancel=False):
     def delete(self, *namespaces):
         all_page = navigate_to(self.parent, 'Details')
         namespaces = list(namespaces)
+        parents = set()
```

Comment:
`parents = {namespace.parent for namespace in namespaces}` 

set comprehensions are a win